dbms/src/Storages/StorageDistributed.cpp

```diff
@@ -399,7 +399,7 @@ BlockInputStreams StorageDistributed::read(
             if (force)
             {
                 std::stringstream exception_message;
-                if (has_sharding_key)
+                if (!has_sharding_key)
```

Comment:
Let's push smaller fixes first, because (after one minute looking) I'm not sure is it acceptable to throw/catch exceptions in non exceptional case.

Reply:
Indeed, they looks really haskish
Tiny fixes is in another pr #8908

Reply:
Will copy style fixes there

Reply:
Oh, no need to (they do not include this lines)
Also note that #8908 has one unique patch